apps/web/src/pages/auth/components/QuestionnaireForm.tsx

```diff
@@ -0,0 +1,214 @@
+  Select,
+  Bell,
+  DelayAction,
+  Digest,
+  Translation,
+  MultiChannel,
+  inputStyles,
+} from '@novu/design-system';
+
+import { api } from '../../../api/api.client';
+import { useAuthContext } from '../../../components/providers/AuthProvider';
+import { useVercelIntegration, useVercelParams } from '../../../hooks';
+import { ROUTES } from '../../../constants/routes.enum';
+import { DynamicCheckBox } from './dynamic-checkbox/DynamicCheckBox';
+
+type Props = {};
+
+export function QuestionnaireForm({}: Props) {
+  const [loading, setLoading] = useState<boolean>();
+
+  const navigate = useNavigate();
+  const { setToken, token } = useAuthContext();
+  const { startVercelSetup } = useVercelIntegration();
+  const { isFromVercel } = useVercelParams();
+
+  const { mutateAsync: createOrganizationMutation } = useMutation<
+    { _id: string },
+    { error: string; message: string; statusCode: number },
+    ICreateOrganizationDto
+  >((data: ICreateOrganizationDto) => api.post(`/v1/organizations`, data));
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    control,
+  } = useForm<IOrganizationCreateForm>({});
+
+  useEffect(() => {
+    if (token) {
+      const userData = decode<IJwtPayload>(token);
+
+      if (userData.environmentId) {
+        if (isFromVercel) {
+          startVercelSetup();
+
+          return;
+        }
+        navigate(ROUTES.HOME);
+      }
+    }
+  }, [token, navigate, isFromVercel, startVercelSetup]);
+
+  async function createOrganization(data: IOrganizationCreateForm) {
+    const { organizationName, ...rest } = data;
+    const createDto: ICreateOrganizationDto = { ...rest, name: organizationName };
+    const organization = await createOrganizationMutation(createDto);
+    const organizationResponseToken = await api.post(`/v1/auth/organizations/${organization._id}/switch`, {});
+
+    setToken(organizationResponseToken);
+  }
+
+  function jwtHasKey(key: string) {
+    if (!token) return false;
+    const jwt = decode<IJwtPayload>(token);
+
+    return jwt && jwt[key];
+  }
+
+  const onCreateOrganization = async (data: IOrganizationCreateForm) => {
+    if (!data?.organizationName) return;
+
+    setLoading(true);
+
+    if (!jwtHasKey('organizationId')) {
+      await createOrganization({ ...data });
+    }
+
+    setLoading(false);
+    if (isFromVercel) {
+      startVercelSetup();
+
+      return;
+    }
+    navigate(ROUTES.GET_STARTED);
+  };
+
+  return (
+    <form noValidate name="create-app-form" onSubmit={handleSubmit(onCreateOrganization)}>
+      <Controller
+        name="jobTitle"
+        control={control}
+        render={({ field }) => {
+          return (
+            <Select
+              label="Job title"
+              data-test-id="questionnaire-job-title"
+              required
+              error={errors.jobTitle?.message}
+              {...register('jobTitle', {
+                required: 'Please input your job title',
+              })}
```

Comment:
If you are using the Controller you don't need the register as well. For the error message you can use 
```
<Controller ...
rules={{
          required: 'Required - Job title',
        }}
render = ...

```